python/python/lance/dataset.py

```diff
@@ -1631,6 +1634,58 @@ def create_index(
             raise NotImplementedError(
                 f"Only {valid_index_types} index types supported. " f"Got {index_type}"
             )
+        if index_type != "IVF_PQ" and one_pass_ivfpq:
+            raise ValueError(
+                f'one_pass_ivfpq requires index_type="IVF_PQ", got {index_type}'
+            )
+
+        # Handle timing for various parts of accelerated builds
+        timers = {}
+        if one_pass_ivfpq and accelerator is not None:
```

Comment:
nit: let's extract this routine to a separate function?

Reply:
I'm staring at this right now and imo this is gonna become an even bigger mess than it already is if I do that. Unfortunately, the way this function is set up right now, it uses a large number of variables to hold state. So if we separated it out, we'd be passing some enormous list of params, and returning a giant tuple of 7 things (if we include the timers, which would also have to be combined to be clean). This whole function should probably be refactored at some point once we finish up adding some of these features.